Toolkit/ArcGISToolkit/PopupController.swift

```diff
@@ -0,0 +1,370 @@
+            ft.applyEdits { (results, error) in
+                
+                if let error = error{
+                    // In this case it is a service level error
+                    print("error applying edits: \(error)")
+                }
+                
+                if let results = results{
+                    
+                    let editErrors = results.flatMap({ self.checkFeatureEditResult($0) })
+                    if editErrors.isEmpty{
+                        print("applied all edits successfully")
+                    }
+                    else{
+                        // These would be feature level edit errors
+                        print("apply edits failed: \(editErrors)")
+                    }
+                    
+                }
+            }
+        }
+        
+        // reset flag
+        addingNewFeature = false
+    }
+    
+    /// This pulls out any nested errors from a feature edit result
+    private func checkFeatureEditResult(_ featureEditResult: AGSFeatureEditResult) -> [Error]{
+        var errors = [Error]()
+        if let error = featureEditResult.error{
+            errors.append(error)
+        }
+        errors.append(contentsOf: featureEditResult.attachmentResults.compactMap({ $0.error }))
+        return errors
+    }
+    
+    public func popupsViewController(_ popupsViewController: AGSPopupsViewController, didCancelEditingFor popup: AGSPopup) {
+        // geometry editing has ended
+        self.sketchEditor.stop()
+        
+        if addingNewFeature{
+            // if was adding new feature, then hide the popup, don't show viewing mode
+            self.cleanupLastPopupsViewController()
+        }
+        
+        // reset flag
+        addingNewFeature = false
+    }
+    
+    public func popupsViewControllerDidFinishViewingPopups(_ popupsViewController: AGSPopupsViewController) {
+        self.cleanupLastPopupsViewController()
+    }
+    
+}
+
+extension PopupController: TemplatePickerViewControllerDelegate {
+    
+    public func templatePickerViewControllerDidCancel(_ templatePickerViewController: TemplatePickerViewController) {
+        templatePickerViewController.dismiss(animated: true)
+    }
+    
+    public func templatePickerViewControllerDidSelectTemplate(_ templatePickerViewController: TemplatePickerViewController, featureTemplateInfo: FeatureTemplateInfo) {
+        templatePickerViewController.dismiss(animated: true){
+            if let feature = featureTemplateInfo.featureTable.createFeature(with: featureTemplateInfo.featureTemplate){
+                self.addingNewFeature = true
+                let popup = AGSPopup(geoElement: feature, popupDefinition: featureTemplateInfo.featureLayer.popupDefinition)
+                self.showPopups([popup])
+                // NOTE: This works around a bug where editing doesn't start until the view is loaded
+                _ = self.popupsViewController?.view
```

Comment:
I missed this. Two things:

1. The modern way of doing this is to call `UIViewController.loadViewIfNeeded()`.
2. I think that bug has been fixed.